packages/core/types/index.d.ts

```diff
@@ -2774,6 +2774,24 @@ export namespace Asciidoctor {
      */
     getRole(): string | undefined;
 
+    /**
+     * Sets the value of the role attribute on this node.
+     *
+     * @param names - A single role name, a space-separated String of role names or a list of role names
+     *
+     * @returns the value of the role attribute.
+     */
+    setRole(...names: string[]): string;
+
+    /**
+     * Sets the value of the role attribute on this node.
+     *
+     * @param names - an Array of role names
+     *
+     * @returns the value of the role attribute.
+     */
+    setRole(names: string[]): string;
```

Comment:
@bitPogo Is it the right to do it? I want to allow the following:

```js
block.setRole('afterword');
block.setRole('lastword', 'closing');
block.setRole(['finalword', 'conclude']);
```

I tried to declare only `setRole(...names: string[]): string;` but the following is invalid:

```js
block.setRole(['finalword', 'conclude'])
```